tensorflow_datasets/core/utils/py_utils.py

```diff
@@ -275,3 +275,9 @@ def rgetattr(obj, attr, *args):
   def _getattr(obj, attr):
     return getattr(obj, attr, *args)
   return functools.reduce(_getattr, [obj] + attr.split("."))
+
+
+def free_disk_size(directory):
+  stat = os.statvfs(directory)
```

Comment:
does this work on Windows and Linux?
maybe we should wrap it in a try/except and if it fails, return `float("inf")`
i think that would also handle the case of cloud filesystems

Reply:
can I use psutil package? It's work on windows and linux. 
I asked because I don't want to add more package to library.

Reply:
psutil also have free disk usage method. `psutil.disk_usage(".").free` 
[link](https://stackoverflow.com/a/29944093/8401425)

Reply:
I fixed like that [pr](https://github.com/tensorflow/datasets/pull/172/commits/7e38b62c838bfdbad351c3cab43551c981cad6b5)

Reply:
Yeah, let's go with psutil. Fine to add as a dep.

Reply:
Should I closed this pr and create a more clean ?